### ISSUE TITLE
MA0003 is reported for array indexer

### ISSUE THREAD

**Issue by drieseng** [meziantou/Meziantou.Analyzer#353]
**Context**
Rule identifier: _MA0003_

**Describe the bug**
When **MA0003** is configured to report a diagnostic for numeric literals, it also reports a diagnostic for array indexers.

**To Reproduce**
Code or steps to reproduce the behavior:

````c#
var d = new[] {"Foo"};

if (d[0] == "X")
{
    d[0] = "XXX";
}
````

**Expected result:**

MA0003 is not reported.

**Actual result:**

MA0003 is reported for both the getter and the setter.

Note:
For **List\<T>** or **Dictionary\<TKey,TValue>** it works fine.

**Comment by meziantou** [comment]
I do not reproduce this issue on my computer / unit tests. Can you provide a full repro (csproj + cs)?

**Comment by drieseng** [comment]
I submitted a PR that adds a test that should allow you to reproduce this issue.

**Comment by meziantou** [comment]
Thanks for the PR! I'm not sure what I've tested to not be able to reproduce the error 🤔

Anyway, it will be fixed in [1.0.710](https://www.nuget.org/packages/Meziantou.Analyzer/1.0.710)!

**Comment by drieseng** [comment]
@meziantou Thanks!!
